libraries/AC_AttitudeControl/AC_AttitudeControl.h

```diff
@@ -222,57 +212,43 @@ class AC_AttitudeControl {
     // Proportional controller with piecewise sqrt sections to constrain second derivative
     static float sqrt_controller(float error, float p, float second_ord_lim);
 
+    // Inverse proportional controller with piecewise sqrt sections to constrain second derivative
+    static float stopping_point(float first_ord_mag, float p, float second_ord_lim);
+
     // User settable parameters
     static const struct AP_Param::GroupInfo var_info[];
 
-protected:
-    // Retrieve a rotation matrix from the vehicle body frame to NED earth frame
-    void get_rotation_vehicle_to_ned(Matrix3f& m);
-
-    // Retrieve a rotation matrix from NED earth frame to the vehicle body frame
-    void get_rotation_ned_to_vehicle(Matrix3f& m);
-
-    // Retrieve a rotation matrix from reference (setpoint) body frame to NED earth frame
-    void get_rotation_reference_to_ned(Matrix3f& m);
+    // calculates the velocity correction form an angle error. The angular velocity has acceleration and
+    // deceleration limits including basic jerk limiting using smoothing_gain
+    float input_shaping_angle(float error_angle, float smoothing_gain, float accel_max, float target_ang_vel);
 
-    // Retrieve a rotation matrix from NED earth frame to reference (setpoint) body frame
-    void get_rotation_ned_to_reference(Matrix3f& m);
+    // limits the acceleration and deceleration of a velocity request
+    float input_shaping_ang_vel(float target_ang_vel, float desired_ang_vel, float accel_max);
 
-    // Retrieve a rotation matrix from vehicle body frame to reference (setpoint) body frame
-    void get_rotation_vehicle_to_reference(Matrix3f& m);
+    // translates body frame acceleration limits to the euler axis
+    Vector3f euler_accel_limit(Vector3f euler_rad, Vector3f euler_accel);
 
-    // Retrieve a rotation matrix from reference (setpoint) body frame to vehicle body frame
-    void get_rotation_reference_to_vehicle(Matrix3f& m);
+    // get_thrust_heading_rotation - calculates two ordered rotations to move the att_from_quat quaternion to the att_to_quat quaternion.
```

Comment:
I mean, it's the same comment as https://github.com/ArduPilot/ardupilot/pull/4336#discussion_r67615012